### PR TITLE
feat(ui): build LXD project aggregate resources card

### LINKED ISSUE #2228: Update "Virtual functions" section of KVM resources card

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Built LXD project aggregate resources card, which includes:
  - `ProjectResourcesCard` which is the wrapper card component for all the individual resource sections
  - `VfResources` which displays the "Virtual functions" section
  - `VmResources` which displays the "Total VMs" section
- Note that it doesn't fully match the [design](https://app.zeplin.io/project/601454d5d1141e1b6ea271b7/screen/602aaa0cf4fce564ee6b02c5) for two reasons:
  1. The colours are wrong. It was decided to keep the original colours instead of changing them for this card.
  2. The "Virtual functions" section shows one interface per row instead of two since some interface names can be long. The design of this section is likely to be changed soon since it's possible to have any number of interfaces.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the "Resources" tab of a LXD KVM and check that the project resources card (mostly) matches the [design](https://app.zeplin.io/project/601454d5d1141e1b6ea271b7/screen/602aaa0cf4fce564ee6b02c5). 

## Fixes

Fixes #2228 
Fixes #2434 

## Screenshot

Screenshot_2021-04-07 LXD resources bolla bolla MAAS

### CODE REVIEW

File: ui/src/app/kvm/views/KVMDetails/LxdResources/LxdResources.tsx [hunk 1]
```diff
@@ -18,16 +18,19 @@ const LxdResources = ({ id }: Props): JSX.Element => {
...
     podSelectors.getById(state, Number(id))
   );
-
   useWindowTitle(`LXD resources ${pod?.name || ""}`);
 
   if (!!pod) {
     return (
       <>
         <Strip className="u-no-padding--top" shallow>
+          <h4 className="u-sv1">Overall</h4>
```
Comment (Caleb-Ellis) [inline ui/src/app/kvm/views/KVMDetails/LxdResources/LxdResources.tsx:27]: I've moved the headings out of the card components to make it easier to put the card in other places (e.g. in virsh details)

File: ui/src/app/kvm/views/KVMDetails/RamResources/RamResources.tsx [hunk 2]
```diff
@@ -70,66 +70,68 @@ const RamResources = ({
...
-                    <strong
-                      className="p-text--x-small u-text--light"
-                      data-test="page-size"
-                    >
-                      {`(Size: ${memoryWithUnit(hugepages.pageSize)})`}
-                    </strong>
-                  </>
-                )}
-              </td>
+      <div className="ram-resources__table-container">
```
Comment (Caleb-Ellis) [inline ui/src/app/kvm/views/KVMDetails/RamResources/RamResources.tsx:73]: Driveby fix for a bug in Firefox where the table cells would take up extra space. Wrapping the table in a div seems to fix it - tables don't seem to like directly interacting with flexbox.

File: ui/src/app/kvm/views/KVMDetails/StorageResources/StorageResources.tsx [hunk 3]
```diff
@@ -60,7 +60,7 @@ const StorageResources = ({ id }: Props): JSX.Element | null => {
 
           return (
             <div className="storage-resources__pool" key={pool.id}>
-              <div className="storage-resources__pool-name">{pod.name}</div>
+              <div className="storage-resources__pool-name">{pool.name}</div>
```
Comment (Caleb-Ellis) [inline ui/src/app/kvm/views/KVMDetails/StorageResources/StorageResources.tsx:63]: Driveby fix - this is supposed to be the pool name not the pod name

**Review by anthonydillon — APPROVED** [review]
Code and QA :+1:
